packages/protoplugin/src/create-es-plugin.ts

```diff
@@ -45,7 +45,7 @@ interface PluginInit {
    * An optional parsing function which can be used to parse your own plugin
    * options.
    */
-  parseOption?: (key: string, value: string) => void;
+  parseOptions?: (rawOptions: { key: string; value: string }[]) => Options;
```

Comment:
What do you think about adding an optional second argument with the standard options to this function?

```ts
(rawOptions: { key: string; value: string }[], standardOptions?: EcmaScriptPluginOptions) => Options
```

To be fair, I can't think of a good use case right now. We can add it once someone requests it.

---

It would be nice to document what `key` and `value` are, and - most importantly - that a `parseOption` function should raise an error when it does not recognize an option.

We could export a type for the raw options in parameter.ts:

```ts
/**
 * Raw options to parse.
 *
 * For example, if a plugin is run with the options foo=123,bar,baz=a,baz=b
 * the raw options are:
 *
 * ```ts
 * [
 *   { key: "foo", value: "123" },
 *   { key: "bar", value: "" },
 *   { key: "baz", value: "a" },
 *   { key: "baz", value: "b" },
 * ]
 * ```
 *
 * If your plugin does not recognize an option, it must throw an Error in
 * parseOptions.
 */
export type RawPluginOptions = {
  key: string;
  value: string;
}[];
```

WDYT?

Reply:
Added the type, I think we can defer the standardOptions parameter.